main.js

```diff
@@ -1,67 +1,105 @@
+const printErrorMsg = (message) => {
+  const errMsg = document.querySelector('.err-msg');
+  errMsg.style.color = 'red';
+  document.querySelector('.err-msg').textContent = message;
+  setTimeout(() => {
+    document.querySelector('.err-msg').textContent = '';
+  }, 2000);
+};
+
+class Book {
+  constructor(id, title, author) {
+    this.id = id;
+    this.title = title;
+    this.author = author;
+  }
+
+  addBook() {
+    const { id, title, author } = this;
+    const bookObj = { id, title, author };
+    books = JSON.parse(localStorage.getItem('books'));
+    if (title === '' || author === '') {
+      printErrorMsg('Please fill in all the fields');
+    } else if (books !== null) {
+      books.push(bookObj);
+      localStorage.setItem('books', JSON.stringify(books));
+      books = JSON.parse(localStorage.getItem('books'));
+      document.getElementById('title').value = '';
+      document.getElementById('author').value = '';
+    } else {
+      books = [];
+      books.push(bookObj);
+      localStorage.setItem('books', JSON.stringify(books));
+      books = JSON.parse(localStorage.getItem('books'));
+      document.getElementById('title').value = '';
+      document.getElementById('author').value = '';
+    }
+  }
+
+  removeBook() {
+    const { id } = this;
     books = books.filter((book) => {
       if (book.id !== id) {
         return true;
       }
       return false;
     });
     localStorage.setItem('books', JSON.stringify(books));
-    li.remove();
-  });
-};
-
-const printErrorMsg = (message) => {
-  document.querySelector('.err-msg').innerHTML = message;
-  setTimeout(() => {
-    document.querySelector('.err-msg').innerHTML = '';
-  }, 2000);
-};
-
-const addBook = (title, author) => {
-  const id = Date.now();
-  const bookObj = { id, title, author };
-  if (title === '' || author === '') {
-    printErrorMsg('Please fill in all the fields');
-  } else {
-    books.push(bookObj);
-    localStorage.setItem('books', JSON.stringify(books));
-    document.getElementById('title').value = '';
-    document.getElementById('author').value = '';
-    displayBooks(bookObj.id, bookObj.title, bookObj.author);
   }
+}
+
+const displayBook = (id, title, author) => {
+  bookList.classList.add('border');
+  const li = document.createElement('li');
+  li.innerHTML = `<div class="d-flex-only">
+  <h3><q>${title}</q></h3>
+  <span>by</span>
+  <h3>${author}</h3>
+  </div>`;
+  const removeBookBtn = document.createElement('button');
+  removeBookBtn.textContent = 'Remove';
+  li.appendChild(removeBookBtn);
+  bookList.appendChild(li);
+  removeBookBtn.addEventListener('click', () => {
+    const book = new Book(id, title, author);
+    id = removeBookBtn.id;
+    book.removeBook();
+    if (li.previousElementSibling === null && li.nextElementSibling === null) {
+      bookList.classList.remove('border');
+      li.remove();
+    } else {
+      li.remove();
+      bookList.classList.add('border');
+    }
+  });
```

Comment:
[OPTIONAL] You will make your code better if you add the `displayBook` function to the Book class.

Reply:
Well noted. Thanks